dracut/10root-unlock-generator/root-unlock-generator

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+UNIT_DIR="$1"
+
+# On first boot, Ignition will set up encryption for the ROOT partition, if configured, and leave it unencrypted.
+# Thus, we do not need systemd-cryptsetup unit to decrypt the partition during the first boot.
+if grep -q "flatcar.first_boot" /proc/cmdline ; then
```

Comment:
Do we need this or is this precaution? I would think that the generator runs before Ignition formatted the disk, so that it wouldn't detect it anyway, or?

Reply:
When we move this shell logic into a helper service, we can let the service be wanted by `ignition-subsequent.target` so that it only runs on second boot. This allows us to drop the duplication of the first-boot cmdline detection.

Reply:
Makes sense. In the updated version, the service is `WantedBy=ignition-subsequent.target`.